java/core/src/java/org/apache/orc/impl/filter/FilterFactory.java

```diff
@@ -184,4 +186,24 @@ static List<BatchFilter> findPluginFilters(String filePath, Configuration conf)
     }
     return filters;
   }
+
+  /**
+   * Delete filter(s) which is not in the allowlist.
+   *
+   * @param filters fully filter list we load from class path.
+   * @param allowList filter allowList. The plugin service will not be
+   *                  loaded if it is not in allowList.
+   */
+  static void deleteFilterNotInAllowList(List<BatchFilter> filters, List<String> allowList) {
+    if (allowList != null && allowList.size() > 0) {
```

Comment:
Given the function name, `deleteFilterNotInAllowList` and description, this logic looks wrong to me.